client/txn.go

```diff
@@ -666,7 +666,12 @@ func (txn *Txn) send(ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.
 	if elideEndTxn && pErr == nil {
 		// Check that read only transactions do not violate their deadline.
 		if endTxnRequest.Deadline != nil {
-			if endTxnRequest.Deadline.Less(txn.Proto.Timestamp) {
+			t := txn.Proto.Timestamp
```

Comment:
Thanks for taking a look. I see. 

I'm still wondering whether we need to do something when `txn.Proto.Timestamp` == `roachpb.ZeroTimestamp` (as the deadline check always passes), but will first come up with more concrete case..